sql/2020/05_Media/meta_open_video_types.sql

```diff
@@ -0,0 +1,49 @@
+#standardSQL
+# meta open graph video types
+
+# returns all the data we need from _almanac
+CREATE TEMPORARY FUNCTION get_meta_og_video_types(almanac_string STRING)
+RETURNS STRUCT<
+  video_types ARRAY<STRING>
+> LANGUAGE js AS '''
+var result = {};
+try {
+    var almanac = JSON.parse(almanac_string);
+
+    if (Array.isArray(almanac) || typeof almanac != 'object') return result;
+
+    for (var node of almanac['meta-nodes']['nodes']) {
+      if (node['property'] === 'og:video') {
+        if (result.video_types == null)
+          result.video_types = [];
+        var url = node['content'];
+        // parse video extension from url
+        var rex = new RegExp('\\.([^/#?]+)[.]([^/#?]+)([#?][^/]*)?$');
```

Comment:
Expand on comment and simplify regex:

```suggestion
        // parse video extension from last part of URL handling any query or hash params
        var rex = new RegExp('([^/]+)[.]([^/]+)([#?][^/]*)?$');
```